packages/sdk38/src/lcdapi/distribution.spec.ts

```diff
@@ -0,0 +1,194 @@
+      const client = new SigningCosmosClient(wasmd.endpoint, faucet.address, wallet, {});
+
+      const chainId = await client.getChainId();
+      const msg = {
+        type: "cosmos-sdk/MsgDelegate",
+        value: {
+          delegator_address: faucet.address,
+          validator_address: validatorAddress,
+          amount: coin(25000, "ustake"),
+        },
+      };
+      const memo = "Test delegation for wasmd";
+      const { accountNumber, sequence } = await client.getNonce();
+      const signBytes = makeSignBytes([msg], defaultFee, chainId, memo, accountNumber, sequence);
+      const signature = await wallet.sign(faucet.address, signBytes);
+      const tx = {
+        msg: [msg],
+        fee: defaultFee,
+        memo: memo,
+        signatures: [signature],
+      };
+
+      const receipt = await client.postTx(tx);
+      if (isPostTxFailure(receipt)) {
+        throw new Error("Delegation failed");
+      }
+
+      await sleep(75); // wait until transactions are indexed
+    }
+  });
+
+  describe("delegatorRewards", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeDistributionClient(wasmd.endpoint);
+      const response = await client.distribution.delegatorRewards(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: {
+          rewards: [
+            {
+              validator_address: validatorAddress,
+              reward: null,
+            },
+          ],
+          total: null,
+        },
+      });
+    });
+  });
+
+  describe("delegatorReward", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeDistributionClient(wasmd.endpoint);
+      const response = await client.distribution.delegatorReward(faucet.address, validatorAddress);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: [],
+      });
+    });
+  });
+
+  describe("withdrawAddress", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeDistributionClient(wasmd.endpoint);
+      const response = await client.distribution.withdrawAddress(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: faucet.address,
+      });
+    });
+  });
+
+  describe("validator", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeDistributionClient(wasmd.endpoint);
+      const response = await client.distribution.validator(validatorAddress);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: {
+          // TODO: This smells like a bug in the backend to me
+          operator_address: Bech32.encode("cosmos", Bech32.decode(validatorAddress).data),
```

Comment:
@ethanfrey? 👆

Reply:
Yes this does seem like they use the wrong bech32 prefix here.